onnxruntime/contrib_ops/cpu/transformers/beam_search_impl_whisper.h

```diff
@@ -0,0 +1,388 @@
+                                                             this->ort_stream_,
+                                                             decoder_subgraph_.UseSequenceAsInputIds(),
+                                                             current_length,
+                                                             cpu_state.sequences,
+                                                             parameters->max_length,
+                                                             decoder_subgraph_.has_decoder_masked_attention_));
+
+    if (decoder_subgraph_.past_present_share_buffer_) {
+      decoder_fetches.reserve(static_cast<int64_t>(decoder_subgraph_.GetFirstPresentOutputIndex()) + 2 * static_cast<int64_t>(decoder_subgraph_.num_layers));
+      decoder_fetches.resize(decoder_subgraph_.GetFirstPresentOutputIndex(), OrtValue());
+      for (int layer = 0; layer < 2 * decoder_subgraph_.num_layers; layer++) {
+        int feed_idx = decoder_subgraph_.GetFirstPastInputIndex() + layer;
+        OrtValue& past_tensor_value = decoder_feeds[feed_idx];
+        Tensor* past_tensor = past_tensor_value.GetMutable<Tensor>();
+        OrtValue present_tensor_value;
+        Tensor::InitOrtValue(past_tensor->DataType(), past_tensor->Shape(), past_tensor->MutableData<T>(),
+                             past_tensor->Location(), present_tensor_value);
+        decoder_fetches.push_back(present_tensor_value);
+      }
+    }
+
+    if (decoder_subgraph_.has_decoder_masked_attention_) {
+      size_t offset = static_cast<size_t>(decoder_subgraph_.GetFirstPastInputIndex());
+      // Here we only need to reorder the past key for self-attention and cross-attention.
+      for (size_t i = 0; i < 2 * static_cast<size_t>(decoder_subgraph_.num_layers); ++i) {
+        ORT_RETURN_IF_ERROR(reorder_past_state_func_(cuda_device_prop_,
+                                                     *decoder_feeds[offset + 2 * i].GetMutable<Tensor>(),
+                                                     beam_state.staging_for_past_state_reorder,
+                                                     this->ort_stream_));
+      }
+      size_t cache_indir_input_offset = static_cast<size_t>(decoder_subgraph_.GetFirstPastInputIndex()) + 4 * static_cast<size_t>(decoder_subgraph_.num_layers) + 2;
+      ORT_RETURN_IF_ERROR(init_cache_indir_func_(*decoder_feeds[cache_indir_input_offset].GetMutable<Tensor>(), this->ort_stream_));
+    }
+  }
+
+  while (current_length < parameters->max_length) {
+    iteration_counter++;
+#ifdef DEBUG_GENERATION
+    auto cur_len = std::to_string(current_length);
+    dumper->Print("***CurrentLength", cur_len, true);
+
+    for (int i = 0; i <= decoder_subgraph_.GetFirstPastInputIndex(); i++) {
+      dumper->Print("decoder_feeds", i, true);
+      dumper->Print("", decoder_feeds[i]);
+    }
+    auto offset = decoder_subgraph_.GetFirstPastInputIndex() + 4 * decoder_subgraph_.num_layers;
```

Comment:
I know this is in the other beam_search_impl file, but I believe these Print() statements throw a segfault when run with DEBUG_GENERATION=1. Worth testing locally with these changes - if this happens should wrap in an `if` statement (or end up deleting as I'm not sure the value in these prints are)

Reply:
The faulty print statements have now been removed

Reply:
Deleted the print statements causing the error